ambry-api/src/main/java/com/github/ambry/router/Router.java

```diff
@@ -63,19 +67,22 @@
    *                       {@link ChunkInfo} object as a source of truth, so the caller should ensure that these
    *                       fields are set accurately.
    * @param callback The {@link Callback} which will be invoked on the completion of the request .
+   * @param quotaChargeEventListener Listener interface to charge quota cost for the operation.
    * @return A future that would contain the BlobId eventually.
    */
   Future<String> stitchBlob(BlobProperties blobProperties, byte[] userMetadata, List<ChunkInfo> chunksToStitch,
-      Callback<String> callback);
+      Callback<String> callback, QuotaChargeEventListener quotaChargeEventListener);
```

Comment:
BTW, I don't think stitchBlob requires a callback here. All the chunk data has already been saved in ambry when we call stitchBlob.

Reply:
stitch blob is charged for only one chunk. Since we create and save the metadata chunk in the stitch blob call, we charge one unit of quota.